lib/ansible/modules/cloud/vmware/vmware_guest_disk.py

```diff
@@ -0,0 +1,604 @@
+                # Select datastore or datastore cluster
+                if 'datastore' in disk:
+                    if 'autoselect_datastore' in disk:
+                        self.module.fail_json(msg="Please specify either 'datastore' "
+                                                  "or 'autoselect_datastore' for disk index [%s]" % disk_index)
+
+                    elif 'autoselect_datastore' not in disk:
+                        # Check if given value is datastore or datastore cluster
+                        datastore_name = disk['datastore']
+                        datastore_cluster = find_obj(self.content, [vim.StoragePod], datastore_name)
+                        if datastore_cluster:
+                            # If user specified datastore cluster so get recommended datastore
+                            datastore_name = self.get_recommended_datastore(datastore_cluster_obj=datastore_cluster)
+                        # Check if get_recommended_datastore or user specified datastore exists or not
+                        datastore = find_obj(self.content, [vim.Datastore], datastore_name)
+                        if datastore is None:
+                            self.module.fail_json(msg="Failed to find datastore named '%s' "
+                                                      "in given configuration." % disk['datastore'])
+                        current_disk['datastore'] = datastore
+                        current_disk['autoselect_datastore'] = False
+                elif 'autoselect_datastore' in disk:
+                    # Find datastore which fits requirement
+                    datastores = get_all_objs(self.content, [vim.Datastore])
+                    if not datastores:
+                        self.module.fail_json(msg="Failed to gather information about"
+                                                  " available datastores in given datacenter.")
+                    datastore = None
+                    datastore_freespace = 0
+                    for ds in datastores:
+                        if isinstance(ds, vim.Datastore) and ds.summary.freeSpace > datastore_freespace:
+                            # If datastore field is provided, filter destination datastores
+                            datastore = ds
+                            datastore_freespace = ds.summary.freeSpace
+                    current_disk['datastore'] = datastore
+
+                if 'datastore' not in disk and 'autoselect_datastore' not in disk:
+                    self.module.fail_json(msg="Either 'datastore' or 'autoselect_datastore' is"
+                                              " required parameter while creating disk for "
+                                              "disk index [%s]." % disk_index)
+
+                if [x for x in disk.keys() if x.startswith('size_') or x == 'size']:
+                    # size_tb, size_gb, size_mb, size_kb, size_b ...?
+                    if 'size' in disk:
```

Comment:
You should document this (currently you only talk about size_tb, size_gb…).

Reply:
OK.